src/passes/literalExpressionEvaluator/literalExpressionEvaluator.ts

```diff
@@ -38,20 +44,12 @@ export class LiteralExpressionEvaluator extends ASTMapper {
     // It is sometimes possible to avoid any calculation and take the value from the type
     // This is not always possible because boolean literals do not contain their value in the type,
     // and because very large int and rational literals omit some digits
-    const result = createLiteralFromType(node.typeString) || evaluateLiteralExpression(node);
-
-    if (result === null) {
-      this.commonVisit(node, ast);
-    } else if (typeof result === 'boolean') {
-      ast.replaceNode(node, createBoolLiteral(result, ast));
-    } else {
-      const intValue = result.toInteger();
-      if (intValue === null) {
-        throw new TranspileFailedError('Attempted to make node for non-integral literal');
-      }
-
-      ast.replaceNode(node, createNumberLiteral(intValue, ast));
+    let result = createLiteralFromType(node, ast);
+    if (result.evaluation === null) {
+      result = evaluateLiteralExpression(node, ast);
```

Comment:
Could be `let result = createLiteralFromType(node, ast) ?? evaluateLiteralExpression(node, ast)`

Reply:
I don't think you can do that in this case, `result` is of type `NodeEvaluationInfo`, so even if the evaluation is `null`, `result` isn't

Reply:
Oh, you're right :)